doc/developer/tracing.md

```diff
@@ -0,0 +1,231 @@
+# Introduction
+
+[`tracing`] is a set of crates used for _logging_ and _tracing_ of rust
+binaries. It is primarily composed of the [`tracing`] crate, which is the
+core way to interact with the framework, and the [`tracing-subscriber`] crate,
+which provides an API to compose systems that consume tracing data.
+
+
+## Logging
+
+The most basic way to use `tracing` is to use it like other logging crates like `log`,
+which additional semi-structured capabilities.
+
+
+Materialize binaries are setup such that [events] are printed on stderr.
```

Comment:
@benesch do we have plans for a centralized logging infrastructure? If so, it would be nice to add them here.